akka-typed-tests/src/test/scala/akka/typed/cluster/sharding/ClusterShardingApiSpec.scala

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (C) 2017 Lightbend Inc. <http://www.lightbend.com/>
+ */
+
+package akka.typed.cluster.sharding
+
+import akka.typed.ActorSystem
+import akka.typed.cluster.Cluster
+import akka.typed.scaladsl.Actor
+import akka.typed.scaladsl.adapter._
+
+class ClusterShardingApiSpec {
+
+  // Compile only for now
+
+  val system: akka.actor.ActorSystem = ???
+  val typedSystem: ActorSystem[Nothing] = system.toTyped
+  val cluster = Cluster(typedSystem)
+
+  trait EntityProtocol
+
+  case class Add(thing: String) extends EntityProtocol
+
+  case object PassHence extends EntityProtocol
+
+  val entityBehavior =
+    Actor.deferred[EntityProtocol] { _ ⇒
+      var things: List[String] = Nil
+
+      Actor.immutable[EntityProtocol] { (_, msg) ⇒
+        msg match {
+          case Add(thing) ⇒
+            things = thing :: things
+            Actor.same
+
+          case PassHence ⇒
+            Actor.stopped
+        }
+      }
+    }
+
+  val sharding = ClusterSharding(typedSystem)
+    .spawn(
+      entityBehavior,
+      "things-lists",
+      ClusterShardingSettings(typedSystem),
+      maxNumberOfShards = 25,
+      handOffStopMessage = PassHence
+    )
+
+  sharding ! ShardingEnvelope("1", Add("bananas"))
+
+  val entity1 = ClusterSharding.entityRefFor("1", sharding)
```

Comment:
I had a previous comment about that the "lookup" of the EntityRef should be with `entityId` and `shardId`, i.e. two strings. I think that has an advantage for supporting serialization of such refs.

```
ClusterSharding(typedSystem).entityRefFor("1", "things-lists")
```

Reply:
and note that the untyped `ClusterSharding` has a Map of started regions/proxies so it might be possible to use that for the "lookup"

Reply:
Ah I see what you meant back there now, seems nicer indeed, will change

Reply:
Hm... not sure about the shardId+entityId though; won't that be hard for users to figure out?

Reply:
Hm, need to think about this one some more

Reply:
sorry, I didn't mean `shardId`. It should have been `entityType`, the same string that you use when you `spawn`. 
"things-lists" in this example

Reply:
Hah ok, yes that's what I had in mind so was confused by the shardId proposal :-)
Yes that makes sense :)

Reply:
and the order of the params should be `entityType`, `entityId`